### PR TITLE
Adding basic Assemble subcommand

### PR DESCRIPTION
**Pull request by GanShun**
Signed-off-by: Gan Shun Lim <ganshun@gmail.com>

### CODE REVIEW

File: uefi/flash.go [hunk 1]
```diff
@@ -35,6 +37,57 @@ type FlashDescriptor struct {
...
 }
 
+func findSignature(buf []byte) (int, error) {
+	if bytes.Equal(buf[16:16+len(FlashSignature)], FlashSignature) {
+		// 16 + 4 since the descriptor starts after the signature
+		return 20, nil
+	}
+	if bytes.Equal(buf[:len(FlashSignature)], FlashSignature) {
+		// + 4 since the descriptor starts after the signature
+		return 4, nil
```
Comment (rminnich) [inline uefi/flash.go]: is it worth defining a nice named constant?

File: uefi/flash.go [hunk 2]
```diff
@@ -35,6 +37,57 @@ type FlashDescriptor struct {
...
+		return err
+	}
+	fd.DescriptorMapStart = uint(descriptorMapStart)
+
+	// Descriptor Map
+	desc, err := NewFlashDescriptorMap(fd.buf[fd.DescriptorMapStart : fd.DescriptorMapStart+FlashDescriptorMapSize])
+	if err != nil {
+		return err
+	}
+	fd.DescriptorMap = *desc
```
Comment (rminnich) [inline uefi/flash.go]: Why isn't DescriptorMap just a pointer too? just wondering. This deref/copy seems more work than needed.

File: uefi/flash.go [hunk 3]
```diff
@@ -139,6 +201,85 @@ func (f *FlashImage) Extract(dirPath string) error {
...
+	// We need to sort them since a) we don't really know the order until we parse the block numbers
+	// and b) the order may have changed anyway.
+	if !f.IFD.Region.BIOS.Valid() {
+		return nil, fmt.Errorf("no BIOS region: invalid region parameters %v", f.IFD.Region.BIOS)
+	}
+	type region struct {
+		P   *Region
+		buf []byte
+	}
+	regions := make([]region, 0, 0)
```
Comment (rminnich) [inline uefi/flash.go]: I can see the length as 0, but why not make the capacity something reasonable, or just not set it? Or just do
var regions []region
if you're going to make it zero size?
If you do this:
regions := make([]region,0, 32) 
then the append will be pretty efficient until you hit 32. No reallocations until you hit it.

File: uefi/gberegion.go [hunk 4]
```diff
@@ -45,3 +46,14 @@ func (gbe *GBERegion) Extract(parentPath string) error {
 	gbe.ExtractPath, err = ExtractBinary(gbe.buf, dirPath, "gberegion.bin")
 	return err
 }
+
+// Assemble assembles the GBE Region from the binary file.
+func (gbe *GBERegion) Assemble() ([]byte, error) {
+	// We have to trust that the ExtractPath we read from the JSON is correct.
```
Comment (rminnich) [inline uefi/gberegion.go]: I'm getting used to this comment everywhere :-)

File: uefi/flash.go (hunk 3 above)
Comment (GanShun) [reply uefi/flash.go]: ok will do

File: uefi/gberegion.go (hunk 4 above)
Comment (GanShun) [reply uefi/gberegion.go]: ok I'll remove it. :P

File: uefi/flash.go (hunk 2 above)
Comment (GanShun) [reply uefi/flash.go]: ok

File: uefi/flash.go (hunk 1 above)
Comment (GanShun) [reply uefi/flash.go]: I'll do it.

File: uefi/flash.go (hunk 2 above)
Comment (GanShun) [reply uefi/flash.go]: done

File: uefi/flash.go (hunk 3 above)
Comment (GanShun) [reply uefi/flash.go]: There are max 4 regions. I'll make cap 4.

**Pull request merged**